### PR TITLE
[CBW-910] Update terms & conditions screen

### PR DESCRIPTION
**Pull request by milsawicki**
Until now each of the wallet types have had their own T&C’s, and in some of the wallets they have been hardcoded making it troublesome to make updates to the T&Cs. Instead, we decided to make a common T&C that is shared between the browser wallet and the recovery phrase based wallets for iOS and Android.

### CODE REVIEW

File: ConcordiumWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (milsawicki) [inline ConcordiumWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:1]: revert that

File: ConcordiumWallet/Service/Network/ApiConstants.swift [hunk 1]
```diff
@@ -50,7 +50,8 @@ struct ApiConstants {
     static let bakerPool = proxyUrl.appendingPathComponent("/v0/bakerPool")
     static let chainParameters = proxyUrl.appendingPathComponent("/v0/chainParameters")
     static let passiveDelegation = proxyUrl.appendingPathComponent("/v0/passiveDelegation")
-    static let appSettings = proxyUrl.appendingPathComponent("/v1/appSettings")
+    static let appSettings = proxyUrl.appendingPathComponent("/v0/appSettings")
```
Comment (milsawicki) [inline ConcordiumWallet/Service/Network/ApiConstants.swift:53]: revert that change

File: ConcordiumWallet/Views/Login/LoginCoordinator.swift [hunk 2]
```diff
@@ -47,55 +50,65 @@ class LoginCoordinator: Coordinator {
...
         let passwordCreated = dependencyProvider.keychainWrapper().passwordCreated()
-        if passwordCreated {
-            showLogin()
-        } else {    
-            showTermsAndConditionsScreen()
-        }
+        let version = dependencyProvider.storageManager().getLastAcceptedTermsAndConditionsVersion()
+        dependencyProvider.appSettingsService()
+            .getTermsAndConditionsVersion()
+            .sink(receiveError: { _ in
```
Comment (milsawicki) [inline ConcordiumWallet/Views/Login/LoginCoordinator.swift:77]: handle that!!!!!

File: ConcordiumWallet/Views/Login/TermsAndConditionsView/TermsAndConditionsView.swift [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+//  ConcordiumWallet
+//
+//  Created by Milan Sawicki on 12/04/2023.
+//  Copyright © 2023 concordium. All rights reserved.
+//
+
+import SwiftUI
+
+struct TermsAndConditionsView: View {
+    @ObservedObject var viewModel: TermsAndConditionsViewModel
```
Comment (milsawicki) [inline ConcordiumWallet/Views/Login/TermsAndConditionsView/TermsAndConditionsView.swift:12]: use @stateobject !!

File: ConcordiumWalletTests/Coordinators/LoginCoordinatorTests.swift [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+import XCTest
+import SwiftUI
+@testable import Mock
+import Combine
+class LoginCoordinatorTests: XCTestCase {
+    var sut: LoginCoordinator!
+    var termsAndConditionsLink = "http://wallet-proxy.mainnet.concordium.software/v0/termsAndConditionsVersion"
+
+    @MainActor
+    func test_start_new_terms_available_password_not_set() {
```
Comment (milsawicki) [inline ConcordiumWalletTests/Coordinators/LoginCoordinatorTests.swift:18]: test_start_toc_never_accepted_and_password_never_set

File: ConcordiumWalletTests/Coordinators/LoginCoordinatorTests.swift [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        sut.start()
+
+        // then
+        XCTAssertTrue(appSettingsMock.getTermsAndConditionsVersionCalled)
+        XCTAssertEqual(appSettingsMock.getTermsAndConditionsVersionCallsCount, 1)
+        XCTAssertTrue(sut.navigationController.topViewController is UIHostingController<TermsAndConditionsView>)
+    }
+    
+    @MainActor
+    func test_start_password_set_terms_up_to_date() {
```
Comment (milsawicki) [inline ConcordiumWalletTests/Coordinators/LoginCoordinatorTests.swift:48]: test_start__password_set_toc_not_changed_skips_toc_view

**Review by bisgardo — APPROVED** [review]

File: ConcordiumWallet/Service/Network/ApiConstants.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (bisgardo) [reply ConcordiumWallet/Service/Network/ApiConstants.swift]: already fixed

**Pull request merged**